File: src/Chess.Api.Request/Requests/RegisterRequest.cs
namespace Chess.Api.Request;

public record RegisterRequest : Request
{
	public RegisterRequest(SessionIdRequest sessionId, string whitePlayerName, string blackPlayerName)
	{
		this.SessionId = sessionId;
		this.WhitePlayerName = whitePlayerName;
		this.BlackPlayerName = blackPlayerName;
	}

	public override SessionIdRequest SessionId { get; }
	public string WhitePlayerName { get; }
	public string BlackPlayerName { get; }
}

File: src/Chess.Angular/ClientApp/src/app/Request/RegisterRequest.ts
import { SessionIdRequest } from "./SessionIdRequest";
import { Request } from "./Request";


export class RegisterRequest implements Request {
	//new(sessionId: SessionIdRequest, whitePlayerName: string, blackPlayerName: string): RegisterRequest;
	constructor(public sessionId: SessionIdRequest, public whitePlayerName: string, public 	blackPlayerName: string) {
		
	}

	requestType: string = "RegisterRequest";
}
